pkg/packagekit/updates.jsx

```diff
@@ -935,9 +887,9 @@ class OsUpdates extends React.Component {
             });
 
             if (this.state.loadPercent)
-                return <Progress value={this.state.loadPercent} title={STATE_HEADINGS[state]} />;
+                return <Progress value={this.state.loadPercent} title={STATE_HEADINGS[this.state.state]} />;
```

Comment:
Oops! Looks like the page actually crashed before, as `state` was not defined? Smells like a test gap. `loading` and `locked` don't have a percentage, but `refreshing` ought to?

Reply:
I will make a test for that, but Loading and locked do not have it since those operations are really fast. I tried it out with a percentage bar and it was quite a fast blink on the screen (which doesn't make it look good in my opinion), so for an operation which is most of the time quite fast, I would just keep the spinner

Reply:
That's right -- we can block a package upgrade (search for `install_lockfile` in [check-packagekit](https://github.com/cockpit-project/cockpit/blob/master/test/verify/check-packagekit)), but I don't know a way how to block the metadata loading. So let's ignore that for now.

Reply:
Oh no I see never crashed. Before it was in Header where there was a 'state' variable. I just accidentally took that variable. But good point on the fact that it's not really tested. The whole functionality where user can click on a button to refresh the available packages to see if something is missing is not tested (since in tests we always add a package before opening browser so it's already present after opening the updates page). So I will add test for that.